cirq-core/cirq/ops/raw_types.py

```diff
@@ -483,10 +483,62 @@ def _commutes_(self, other: Any, *, atol: float = 1e-8) -> None | NotImplemented
 
     def _mul_with_qubits(self, qubits: tuple[cirq.Qid, ...], other):
         """cirq.GateOperation.__mul__ delegates to this method."""
+        if isinstance(other, Operation):
+            try:
+                # Try using pauli expansion if both operations have single-item expansions
+                pauli_expansion_self = protocols.pauli_expansion(self.on(*qubits))
+                pauli_expansion_other = protocols.pauli_expansion(other)
+
+                if (
+                    pauli_expansion_self is not None
+                    and len(pauli_expansion_self) == 1
+                    and pauli_expansion_other is not None
+                    and len(pauli_expansion_other) == 1
+                ):
+
+                    gate_self, coef_self = next(iter(pauli_expansion_self.items()))
+                    gate_other, coef_other = next(iter(pauli_expansion_other.items()))
+
+                    from cirq.ops.pauli_string import PauliString
+
+                    return (
+                        coef_self
+                        * PauliString({q: gate_self for q in qubits})
+                        * coef_other
+                        * PauliString({q: gate_other for q in qubits})
+                    )
+            except TypeError:
+                return NotImplemented
         return NotImplemented
 
     def _rmul_with_qubits(self, qubits: tuple[cirq.Qid, ...], other):
         """cirq.GateOperation.__rmul__ delegates to this method."""
+        if isinstance(other, Operation):
+            try:
+                # Try using pauli expansion if both operations have single-item expansions
```

Comment:
I think this logic could just be moved to `_try_interpret_as_pauli_string` so it's not duplicated.